lib/job.js

```diff
@@ -204,21 +204,26 @@ Job.prototype.promote = function(){
     });
 };
 
+/**
+ * Attempts to retry the job. Only a job that has failed can be retried.
+ *
+ * @return {Promise} If resolved and return code is 1, then the queue emits a waiting event
+ * otherwise the operation was not a success and throw the corresponding error. If the promise
+ * rejects, it indicates that the script failed to execute
+ */
 Job.prototype.retry = function(){
```

Comment:
yes a retry with parameters, or the script part with parameters and then reuse that code in the repeat function.